eventbridge/src/main/scala/uk/gov/nationalarchives/DAEventBridgeClient.scala

```diff
@@ -59,7 +44,24 @@ object DAEventBridgeClient:
     .credentialsProvider(DefaultCredentialsProvider.create())
     .build()
 
-  def apply[F[_]: Async](eventBridgeAsyncClient: EventBridgeAsyncClient) =
-    new DAEventBridgeClient[F](eventBridgeAsyncClient)
+  def apply[F[_]: Async](eventBridgeAsyncClient: EventBridgeAsyncClient): DAEventBridgeClient[F] =
+    new DAEventBridgeClient[F] {
+      override def publishEventToEventBridge[T, U](sourceId: String, detailType: U, detail: T)(using
+          enc: Encoder[T]
+      ): F[PutEventsResponse] = {
+        val detailAsString = detail.asJson.printWith(Printer.noSpaces)
+        val requestEntry: PutEventsRequestEntry = PutEventsRequestEntry.builder
+          .detail(detailAsString)
+          .source(sourceId)
+          .detailType(detailType.toString)
+          .build()
+        val putEventsRequest = PutEventsRequest.builder
+          .entries(requestEntry)
+          .build()
+
+        Async[F].fromCompletableFuture(Async[F].pure(eventBridgeAsyncClient.putEvents(putEventsRequest)))
+      }
+
+    }
 
-  def apply[F[_]: Async]() = new DAEventBridgeClient[F](asyncClient)
+  def apply[F[_]: Async](): DAEventBridgeClient[F] = DAEventBridgeClient[F](asyncClient)
```

Comment:
Do we need this? could we not just add a default argument to line 42, e.g. `def apply[F[_]: Async](eventBridgeAsyncClient: EventBridgeAsyncClient=asyncClient): DAEventBridgeClient[F]`?

Reply:
Yeah, ok. I've changed it.